plugins/kubernetes/app/models/kubernetes/release_doc.rb

```diff
@@ -64,6 +64,10 @@ def blue_green_color
       kubernetes_release.blue_green_color if kubernetes_role.blue_green?
     end
 
+    def deploy_group
```

Comment:
kinda breaks the expectation of soft_deletion and might break eager loading too ...
prefer to either support nils or delete dependencies when nils are not supported

Reply:
This is not breaking the eager loading, I just checked.
This also triggering the expectation of soft_deletion, I just checked by deleting the 'deleted_at' from table.
Here, I only try to support nils, There are lots of places we need to fix to support nils, I think this is the easiest and safest way to support nils.

Reply:
I'd prefer a pattern that can be used for everything so it's no guesswork on how associations will behave.
Adding with_deleted for this 1 association might mean less changes are necessary right now, but it makes things inconsistent / adds tech-debt. 

Maybe as minimal compromise make all deploy-group belongs-to use with_deleted ?
(search for with_deleted to remove existing with_deleted calls that will become obsolete and add a cleanliness test that checks for belongs_to :deploy_group + def deploy_group)

Changing all belongs_to sounds kinda hacky/dangerous.

Using with_deleted in views as needed is hacky/1-off too, but at least consistent.

Reply:
The following are the places needs the change:
1) /plugins/kubernetes/app/views/samson_kubernetes/_deploy_tab_body.html.erb
2) /plugins/kubernetes/app/views/kubernetes/releases/show.html.erb
3) /plugins/kubernetes/app/models/kubernetes/template_filler.rb
4) /plugins/kubernetes/app/models/kubernetes/release.rb
5) /plugins/kubernetes/app/models/kubernetes/deploy_executor.rb
 May be more..